PublishParasoftResults/ParaReportPublishService.ts

```diff
@@ -440,18 +457,34 @@ export class ParaReportPublishService {
         return node.attributes.hasOwnProperty('toolName') && node.attributes['toolName'] == 'C++test' && !node.attributes.hasOwnProperty('prjModule');
     }
 
-    processResults = (inputReportFiles: string[], index: number): void =>{
+    processResults = async (inputReportFiles: string[], index: number): Promise<void> =>{
         if (index < inputReportFiles.length - 1) {
-            this.transformReports(inputReportFiles, ++index);
+            await this.transformReports(inputReportFiles, ++index);
         } else {
             if (this.xUnitReports.length > 0) {
                 let tp: tl.TestPublisher = new tl.TestPublisher('JUnit');
                 tp.publish(this.xUnitReports, this.mergeResults, this.platform, this.config, this.testRunTitle, this.publishRunAttachments);
             }
             if (this.sarifReports.length > 0) {
-                for (var i = 0; i < this.sarifReports.length; ++i) {
-                    this.checkAndAddUnbViolIdForSarifReport(this.sarifReports[i]);
-                    tl.uploadArtifact("Container", this.sarifReports[i], "CodeAnalysisLogs");
+                let referenceSarifReports: FileEntry[] = [];
+                if (this.referenceBuild == this.buildNumber) {
+                    tl.warning(`All issues are treated as new since the build ID set is the same as the current build`);
```

Comment:
How about: Using the current build as the reference, all issues will be treated as new.